src/QsCompiler/CompilationManager/ContextBuilder.cs

```diff
@@ -31,7 +32,7 @@ internal static void VerifyTokenOrdering(IEnumerable<CodeFragment> tokens)
                 throw new ArgumentNullException(nameof(tokens));
             }
 
-            Position previousEnding = null;
+            Lsp.Position previousEnding = null;
```

Comment:
Since the Protocol namespace is already opened, why do you need to qualify `Position` with `Lsp`? Is it conflicting with something? If it is just to make the code easier to read, then I might suggest adding the qualification to the `Diagnostic` type references in this file as well and removing line 13 which opens the namespace.

Reply:
Without `Lsp`, the reference is ambiguous, because `Microsoft.Quantum.QsCompiler.DataTypes.Position` is also in scope. I think I like the idea of qualifying all LSP types, even unambiguous ones like `Diagnostic`, though.

Reply:
If I consistently apply the rule that `Microsoft.VisualStudio.LanguageServer.Protocol` should always be opened with an alias to all the files changed in this PR, though, there will be a lot of changes, so it is probably better to only use `Lsp.` for ambiguous types for now.

Reply:
Ah okay so it _does_ resolve an ambiguity. Makes sense.